starter-code/src/functions-and-arrays.js

```diff
@@ -1,16 +1,90 @@
 // Iteration #1: Find the maximum
+function maxOfTwoNumbers(num1, num2) {
+  if (num1 > num2) {
+    return num1
+  } else if (num1 < num2) {
+    return num2
+  } else if (num1 = num2) {
+    return num1
+  }
+};
+
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
+function findLongestWord(words) {
+  let longestWord = '';
+  if (words.length == 0) {
+    return null
+  } else if (words.length == 1) {
+    return words[0]
+  } else if (words.length > 1) {
+    words.forEach(element => {
+      if (element.length > longestWord.length) {
+        longestWord = element;
+      }
+    });
+  } return longestWord;
+};
+
+
 // Iteration #3: Calculate the sum
 
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
+let isAllZero = numbers.every(item => item === 0);
+
+let sumNumArray = 0;
+
+function sumNumbers(numbers) {
+  if (numbers.length === 0) {
+    return 0;
+  } else if (numbers.length == 1) {
+    return numbers[0];
+  } else if (numbers.length > 1) {
+    if (isAllZero) {
+      return 0;
+    } else {
+      for (let i = 0; i < numbers.length; i++) {
+        sumNumArray += numbers[i];
+      } return sumNumArray;
+    }
+  }
+};
+
+// Bonus 3.1 
+
+const mixedArr = [6, 12, 'miami', 1, true, 'barca', '200', 'lisboa', 8, 10];
+
+sumArr = 0;
+
+let dataType = '';
+if (mixedArr.every(item => item === 0)) {
+  dataType = 0;
+}
+
+function sum(mixedArr) {
```

Comment:
te comparto el código de la función usamos typeof para saber el tipo del elemento que se evalúa 
y para arrojar un error usamos throw y new Error('mensaje') porque así crearemos el error para mostrar
`
const sum = (arr) => {
  let numSum = 0

if (arr.length !== 0) {
    for (number in arr) {
      if(typeof arr[number] === 'object'){
        throw new Error("Unsupported data type sir or ma'am")
      }
      if (typeof arr[number] == "number") {
        numSum += arr[number]
      } else if (typeof arr[number] == "string") {
        numSum += arr[number].length
      } else if (typeof arr[number] == "boolean") {
        numSum += Number(arr[number])
      }
    }
    return numSum


  } else {
    return numSum
  }
}
`